ui/app.py

```
@@ -1,168 +1,191 @@
+            st.json(message["content"])
+            # Add download button for the report
+            pdf_report = create_pdf_report(message["content"])
+            st.download_button(
+                label="Download Report as PDF",
+                data=pdf_report,  # Read the PDF content
+                file_name="research_report.pdf",
+                mime="application/pdf",
+                key=f"download_{id(message)}" # Use a unique key
+            )
+        elif message.get('type') == 'thinking':
+             st.info(message["content"])
```

Flake8 (7.2.0)

[error] indentation is not a multiple of 4

(E111)